### PR TITLE
[cadl-typescript] Update to cadl 0.39.0

### PR DESCRIPTION
**Pull request by joheredi**
Updating @cadl-lang/compiler to 0.39.0 and other dependencies

### CODE REVIEW

**Review by deyaaeldeen — APPROVED** [review]

File: packages/cadl-rlc-test/package.json [hunk 1]
```diff
@@ -3,16 +3,16 @@
...
-    "@cadl-lang/openapi3": "0.38.0",
-    "@cadl-lang/prettier-plugin-cadl": "0.38.0",
+    "@cadl-lang/rest": "~0.39.0",
+    "@cadl-lang/openapi": "~0.39.0",
+    "@cadl-lang/versioning": "~0.39.0",
+    "@azure-tools/cadl-azure-core": "~0.25.0",
+    "@azure-tools/cadl-autorest": "~0.24.0",
+    "@azure-tools/cadl-dpg": "~0.25.0",
+    "@cadl-lang/openapi3": "~0.38.0",
+    "@cadl-lang/prettier-plugin-cadl": "~0.38.0",
```
Comment (qiaozha) [inline packages/cadl-rlc-test/package.json:15]:
```suggestion
    "@azure-tools/cadl-autorest": "~0.25.0",
    "@azure-tools/cadl-dpg": "~0.25.0",
    "@cadl-lang/openapi3": "~0.39.0",
    "@cadl-lang/prettier-plugin-cadl": "~0.39.0",
```

File: packages/cadl-rlc-test/eng/smoke-test.js [hunk 2]
```diff
@@ -15,7 +15,8 @@ function generate(path) {
 }
 
 function build(path) {
-  const command = `cd ${path}/cadl-output && npm install && npm run build`;
+  // Remove --ignore-engines when https://github.com/Azure/autorest.typescript/issues/1733 is fixed
```
Comment (qiaozha) [inline packages/cadl-rlc-test/eng/smoke-test.js:18]: wonder if we still need this ?

**Comment by qiaozha** [comment]
I don't see any changes related with templateArguments. see the change log here https://microsoft.github.io/cadl/release-notes/release-2023-01-12#api-templated-operation-inside-interface

**Comment by MaryGao** [comment]
@joheredi The conflict could be resolved once you re-generate the openai folder in local.

**Pull request merged**